src/main/scala/cromwell/engine/workflow/ValidateActor.scala

```diff
@@ -10,65 +15,181 @@ import wdl4s._
             StatusCodes.OK,
             APIResponse.success("Validation succeeded."))
 
+        case Success(_) =>
+          val msg = "No backend available that can accept this workflow!"
+          logger.info(msg)
+          requester ! RequestComplete(
+            StatusCodes.BadRequest,
+            APIResponse.fail(new IllegalStateException(msg)))
+
         case Failure(ex) =>
           val messageOrBlank = Option(ex.getMessage).mkString
-          logger.info(s"$tag error $sentBy: $messageOrBlank")
-          sentBy ! RequestComplete(
+          logger.info(s"$tag error $requester: $messageOrBlank")
+          requester ! RequestComplete(
             StatusCodes.BadRequest,
             APIResponse.fail(ex))
       }
-    }
+
+      //This message initiates the Validation of the workflow. It will:
+      // 1.) Try to build a NamespaceWithWorkflow object
+      // 2.) Try to coerce the inputs
+      // 3.) Find Backends that can accept this worklfow (fails this hurdle if there is no backend found)
+    case RequestValidation =>
+      val requster = sender()
+      val futureValidationOutcome: Future[ValidationSuccess] = for {
+        namespaceWithWorkflow <- Future(NamespaceWithWorkflow.load(wdlSource))
+        coercedInputs <- buildCoercedInputs(namespaceWithWorkflow, workflowInputs)
+        validatedBackends <- executableBackendsForWf(namespaceWithWorkflow, Option(coercedInputs), workflowOptions)
+      } yield ValidationSuccess(namespaceWithWorkflow,
+        Option(coercedInputs),
+        workflowOptions,
+        validatedBackends,
+        Option(WorkflowSourceFiles(wdlSource, workflowInputs.getOrElse("{}"), workflowOptions.getOrElse("{}"))))
+
+      futureValidationOutcome onComplete {
+        case Success(validationOutcome: ValidationSuccess) =>
+          requster ! validationOutcome
+        case Failure(reason) =>
+          requster ! ValidationFailure(reason)
+      }
+  }
+
+  private def buildCoercedInputs(namespaceWithWorkflow: NamespaceWithWorkflow, rawInputs: Option[WdlJson]): Future[WorkflowCoercedInputs] = {
+    val inputs = workflowInputs.get.parseJson.asJsObject.fields
+    Future.fromTry(namespaceWithWorkflow.coerceRawInputs(inputs))
   }
 
+  /**
+    * This method will validate the workflow with all the available backends. The WF fails
+    * validation if this method returns a list of size 0, which meant that no backend was willing to accept
+    * this workflow.
+    * Later on, this method can be changed to a 'per-call-basis' acceptance test by the backends,
+    * i.e. it may return a Map of [Call -> ExecutableBackends]
+    *
+    * @param namespace     workflow with the namespaces already resolved
+    * @param wfInputs      Already coerced inputs
+    * @param wfOptionsJson workflow options
+    * @return List of BackendConfigurationEntry objects, representing which backends can accept this WF
+    */
+  private def executableBackendsForWf(namespace: NamespaceWithWorkflow,
```

Comment:
Name of the method is not intuitive, use proper semantics  as per your comment. It should be something like validationOfExecutableBackends or(validationOfPluggableBackends). Second break this method in two parts 1) proper validation for backends that have validation class 2) another method bringing will execute backends.